(hack) Add debugging guide (#2871)
puppetlabs/bolt: guides/debugging.txt

TOPIC
    debugging

DESCRIPTION
    When Bolt isn't behaving as expected, there are a few helpful commands and
    logs that can help identify common issues. The first place to look is in
    `<PROJECT>/bolt-debug.log`, which contains debug-level logs from the last Bolt
    run. This log file includes where the Bolt project was loaded from, the
    location of any configuration or inventory files that were loaded, and the
    modulepath that modules were loaded from.

    If you're having issues with loading targets or target configuration, you

>     can see the list of resolved Bolt target names by running 'bolt inventory

Reviewer: ```suggestion
    can see the list of resolved Bolt target names by running `bolt inventory
```

>     can see the list of resolved Bolt target names by running 'bolt inventory
>     show' on *nix systems or 'Get-BoltInventory' in PowerShell. To see the

Reviewer: ```suggestion
    can see the list of resolved Bolt target names by running `bolt inventory
    show` on *nix systems or `Get-BoltInventory` in PowerShell. To see the
```

For consistency with the rest of the guide.

>     show' on *nix systems or 'Get-BoltInventory' in PowerShell, and can see the
>     resolved configuration for each target by running with the `--detail` or
>     `-Detail` options.

Reviewer 2: ```suggestion
    show' on *nix systems or 'Get-BoltInventory' in PowerShell. To see the
    resolved configuration for each target, run the command with the `--detail` or
    `-Detail` options.
```
